sim/scenarios/droplist/droplist-error-model.cc

```diff
@@ -0,0 +1,35 @@
+#include <map>
+#include "droplist-error-model.h"
+
+using namespace std;
+
+NS_OBJECT_ENSURE_REGISTERED(DroplistErrorModel);
+ 
+TypeId DroplistErrorModel::GetTypeId(void) {
+  static TypeId tid = TypeId("DroplistErrorModel")
+    .SetParent<ErrorModel>()
+    .AddConstructor<DroplistErrorModel>()
+    ;
+  return tid;
+}
+ 
+DroplistErrorModel::DroplistErrorModel() { }
+
+void DroplistErrorModel::DoReset(void) { }
+ 
+bool DroplistErrorModel::DoCorrupt(Ptr<Packet> p) {
+  static int packet_num = 0;
+  return drops.empty() ? 0 : drops.find(++packet_num) != drops.end();
```

Comment:
I was trying to optimize since I figured that find() might be expensive to run all the time. It's likely that the compiler probably optimizes this case out anyway. Removed.